app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookTHCallback.kt

```diff
@@ -0,0 +1,109 @@
+package com.github.factotum_sdp.factotum.ui.roadbook
+
+import android.annotation.SuppressLint
+import android.app.AlertDialog
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.ItemTouchHelper.*
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import com.github.factotum_sdp.factotum.R
+import com.google.android.material.snackbar.Snackbar
+
+/**
+ * RoadBookTHCallback abstract class :
+ *
+ * ItemTouchHelper callback providing the different RoadBook events on
+ * specific touch gesture events.
+ *
+ * The default gesture detections are :
+ * - UP ans DOWN drag directions
+ * - LEFT and RIGHT directions on a Swipe State Action
+ *
+ * The resulted events are :
+ * - On Drag&Drop :
+ *          RoadBook ViewModel data list reordering with several
+ *          swap actions appearing only in the back-end and
+ *          one final push observable by the front-end when the ViewItem is finally dropped.
+ * - On Swipe Left :
+ *          Deletion of the DestinationRecord swiped through an AlertDialog.
+ * - On Swipe Right :
+ *          Edition of the DestinationRecord swiped through a custom AlertDialog.
+ */
+abstract class RoadBookTHCallback() :
+    SimpleCallback(
+    UP or DOWN,
+    LEFT or RIGHT or ACTION_STATE_SWIPE
+    ) {
+
+    abstract fun getHost(): Fragment
+    abstract fun getRbViewModel(): RoadBookViewModel
+    abstract fun getRecyclerView(): RecyclerView
+
+    override fun onMove(
+        recyclerView: RecyclerView,
+        viewHolder: ViewHolder,
+        target: ViewHolder
+    ): Boolean {
+        try {
+            val fromPosition = viewHolder.absoluteAdapterPosition
```

Comment:
Ideally it would be best to test this function.

Reply:
As explained in the PR this code concerns and many times in the Team meeting it is the few lines that I can't traverse, as the Drag and Drop Gesture can't be simulated on an emulator.